### DISCUSSION TITLE
Installation on ARM architectures

### DISCUSSION THREAD

**Discussion by IchthysMaranathaCopy** [Budibase/budibase#1647] [Help] [answered]
Hi, there's a problem installing on Arm 64 servers such as those available on AWS and Oracle Cloud. ibmcom/couchdb3 does not support the arm arch. Error is <no matching manifest for linux/arm64/v8 in the manifest list entries>

Any workaround installing normal couchdb and changing parameters?

**Comment by IchthysMaranathaCopy** [comment]
Tried with couchdb instead of ibmcom/couchdb3 and installation succeeds. But results in error on portal <upstream connect error or disconnect/reset before headers. reset reason: connection failure>

**Comment by mike12345567** [comment, marked as answer]
Hi @manojmat29 we used to make use of the standard CouchDB image, however we transitioned as of our latest update to use IBMs image to include the new 3.0 Lucene based free text search engine. This is something that the standard image is not bundled with. I expect that it is failing as we heavily utilise the search engine in our latest update.

We would love to support ARM architectures fully but haven't had time to create a custom image, this might also be an issue the IBM team could/might address as Lucene is JVM based and the lack or ARM image may simply be an oversight.

**Comment by IchthysMaranathaCopy** [comment]
Hi, checked the container logs and it seems that the error that is usually thrown up for mismatched arch <exec user process caused: exec format error > is what is logged for worker and app. Also for ibmcom/couchdb3 if it is used instead of couchdb. So, merely getting the couchdb3 image for ARM 64 might not work.
Oracle has been really generous offering a 4 core 24 GB memory ARM 64 instance in its always free tier, so there might be an untapped market there.  Please do consider supporting ARM if feasible. Thanks.

**Comment by mike12345567** [comment]
Hi @manojmat29 - for our worker and server images I don't think it would be very difficult to support ARM since all they are is a slim Linux image with Node, should be easy enough to enable building and deployment of an ARM tagged image as well. The issue still relates back to there being no easy CouchDB 3.0 ARM image available though sadly.

**Reply by IchthysMaranathaCopy** [reply]
Was able to install and run on aarch64
Steps
1 Register qemu-binfmt interpreter for amd64 by running
docker run --rm --privileged aptman/qus -s -- -p x86_64
2 After running budi hosting --init modify yaml to specifically use amd64 couchdb3 image
image: ibmcom/couchdb3@sha256:c039480c5eec2ecf3f96eb48202d7400e40004246ebf7ac922f134d3f9abf251
One can use <docker manifest inspect ibmcom/couchdb3> to identify the sha56 for latest amd64 image
Then with budi hosting --start everything works

**Comment by Rory-Powell** [comment]
@manojmat29 I ran into the same issue when doing first time installation, an alternative workaround is to update the `platform` in the docker-compose file to `linux/amd64`. There's some more details around it in this thread:

https://github.com/Budibase/budibase/issues/1828

**Reply by IchthysMaranathaCopy** [reply]
Thanks Rory but without the emulation set up by docker run --rm --privileged aptman/qus -s -- -p x86_64, it won't work on linux aarch64. You may already have x86_64 emulation enabled in your Mac. 

But yes, setting platform to linux/amd64 in yaml is better than specifically downloading the amd64 image using its sha value.

**Reply by Rory-Powell** [reply]
Thanks for calling that out, that's right I am running rosetta to enable it

**Comment by HyperCrowd** [comment]
Docker doesn't even support the `platform` option anymore, so this solution literally doesn't work.

https://github.com/docker/cli/issues/1050

**Reply by dncpax** [reply]
if you are on arm ubuntu install lxd, create an lxd vm with x86 arch, and then install docker and budi. matrioska virtualization...
